src/hooks/useIntersection.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react';
+const useIntersection = (
+    loadingRef: React.MutableRefObject<HTMLDivElement | null>,
+    bottomBoundaryRef: React.MutableRefObject<HTMLDivElement | null>,
+    onLoadMore: () => void
+) => {
+    const [isIntersecting, setIsIntersecting] = useState(false);
+
+    useEffect(() => {
+        const options = {
```

Comment:
do we need to define the `options` constant in a `useEffect`? It could very well be outside useIntersection.

Reply:
Done